io/include/pcl/io/real_sense_2_grabber.h

```diff
@@ -0,0 +1,171 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2018-, Open Perception, Inc.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#pragma once
+
+#include <pcl/io/boost.h>
+#include <pcl/io/grabber.h>
+
+#include <librealsense2/rs.hpp>
+
+namespace pcl
+{
+  struct PointXYZ;
+  struct PointXYZRGB;
+  struct PointXYZRGBA;
+  struct PointXYZI;
+  template <typename T> class PointCloud;
+
+  /** \brief Grabber for Intel Realsense 2 SDK devices (D400 series)
+  * \author Patrick Abadi <patrickabadi@gmail.com>, Daniel Packard <pack3754@gmail.com>
+  * \ingroup io
+  */
+  class PCL_EXPORTS RealSense2Grabber : public pcl::Grabber
+  {
+  public:
+    /** \brief Constructor
+    * \param[in] serial_number optional to start up a specific device
+    */
+    RealSense2Grabber (const std::string& serial_number = "", const std::string& file_name = "");
```

Comment:
In this implementation, It need to pass two arguments when capturing from file.
I think that this specification is generally not favored.
```cpp
// capture from bag file
boost::shared_ptr<pcl::RealSense2Grabber> grabber = boost::make_shared<pcl::RealSense2Grabber> ("", "file.bag");
```
Is there a special reason for this implementation?
I think it should define two contractors. (https://github.com/PointCloudLibrary/pcl/pull/2214#issuecomment-365936638)

Reply:
Hi @UnaNancyOwen that makes sense since the user can only do one or the other.  I've separated the constructors now.